src/analysis/inner_product_space/adjoint.lean

```diff
@@ -298,6 +298,21 @@ lemma is_adjoint_pair (A : E' →ₗ[ℝ] F') :
 
 end real
 
+lemma gram_self_adjoint (T : E →ₗ[𝕜] E): is_self_adjoint (T.adjoint * T) :=
+begin
+  intros x y,
+  simp only [linear_map.mul_apply, linear_map.adjoint_inner_left, linear_map.adjoint_inner_right],
+end
+
+lemma gram_positive (T : E →ₗ[𝕜] E) :
+∀ (x : E), 0 ≤ is_R_or_C.re ⟪ (T.adjoint * T) x, x ⟫ ∧ is_R_or_C.im ⟪ (T.adjoint * T) x, x⟫ = 0 :=
+begin
+  intro x,
+  rw [linear_map.mul_apply, linear_map.adjoint_inner_left, inner_self_eq_norm_sq_to_K],
+  norm_cast,
+  exact ⟨ sq_nonneg (∥ T x ∥), rfl ⟩,
+end
```

Comment:
I think I would break this down into two lemmas, one for the real part and one for the imaginary part, with names like `re_inner_adjoint_mul_self_nonneg` and `im_inner_adjoint_mul_self_eq_zero`. (The reason is that when you want to use this lemma, you'll always have to unpack the `∧`  which is more annoying than just having two separate lemmas.) Also, you might want to consider putting `T.adjoint * T` on the right-hand side of the inner product, since it's linear in the right-hand term, but this is just esthetics.

Reply:
I will follow through on this suggestion; it makes sense to me.

Reply:
I optimistically gave each lemma an implicit `{x : E}`, but I am not 100% on this versus `(x : E)`.  Do you mind confirming/denying that the current state makes sense?

Reply:
I've fixed this, thanks.  I must have been thinking that inferring from the statement was possible, but I will focus on the relationships between parameters.